### PR TITLE
Added ProgressBar Element

### PR DESCRIPTION
**Pull request by mateosilguero**
Close: #1971

### CODE REVIEW

**Comment by iRoachie** [comment]
Hey @mateosilguero thanks for submitting this!  Do you have an pictures or videos of this usage? When we add a new component we have to add docs and types as well. Here's example of the Button docs https://react-native-elements.github.io/react-native-elements/docs/button.html

**Comment by mateosilguero** [comment]
Hi ! now, i commited docs for this component

File: website/i18n/en.json [hunk 1]
```diff
@@ -287,6 +290,9 @@
       "version-1.2.0/version-1.2.0-pricing": {
         "title": "Pricing"
       },
+      "version-1.2.0/version-1.2.0-progressbar": {
+        "title": "ProgressBar"
+      },
```
Comment (iRoachie) [inline website/i18n/en.json]: Remove these lines

File: src/progressbar/ProgressBar.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    height: PropTypes.number,
+    indeterminate: PropTypes.bool,
+    indeterminateAnimationDuration: PropTypes.number,
+    progress: PropTypes.number,
+    style: PropTypes.any,
+    unfilledColor: PropTypes.string,
+    width: PropTypes.number,
+    useNativeDriver: PropTypes.bool,
+    animationConfig: PropTypes.object,
+    animationType: PropTypes.oneOf(['decay', 'timing', 'spring']),
```
Comment (iRoachie) [inline src/progressbar/ProgressBar.js]: I'm not sure if it needs to be this customizable.

Think we can go with 

```
color
unfilledColor
indeterminate
progress
style
containerStyle // All our components have this
```

File: docs/progressbar.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
+---
+id: progressbar
+title: ProgressBar
+---
+
+Progress bar is an indicator used to present progress of some activity or task.
```
Comment (iRoachie) [inline docs/progressbar.md]:
```suggestion
Progress bars are indicators used to present the progress of some activity or task.
```

File: docs/progressbar.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+## Reference
+
+### `color`
+
+Progress bar color.
+
+|      Type      | Default |
+| :------------: | :-----: |
+| Color (string) | primary |
```
Comment (iRoachie) [inline docs/progressbar.md]:
```suggestion
|  string | primary (theme) |
```

File: docs/progressbar.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+- [`constainerStyle`](#constainerStyle)
+- [`unfilledColor`](#unfilledColor)
+
+---
+
+## Reference
+
+### `color`
+
+Progress bar color.
```
Comment (iRoachie) [inline docs/progressbar.md]:
```suggestion
Fill color of the bar.
```

File: docs/progressbar.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+|      Type      | Default |
+| :------------: | :-----: |
+| Color (string) | primary |
+
+---
+
+### `indeterminate`
+
+prop to indicate if progress will be indeterminate. if true, ignore `progress` prop.
```
Comment (iRoachie) [inline docs/progressbar.md]:
```suggestion
Toggles if progress will be indeterminate (infinite) or determinate. If true, the `progress` prop is not used.
```

File: docs/progressbar.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+---
+
+### `progress`
+
+progress value between 0 and 1.
+
+|  Type  | Default |
+| :----: | :-----: |
+| Number |    0    |
```
Comment (iRoachie) [inline docs/progressbar.md]:
```suggestion
| number |    0    |
```

File: docs/progressbar.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+|        Type         |    Default     |
+| :-----------------: | :------------: |
+| View style (object) | Internal Style |
+
+---
+
+### `unfilledColor`
+
+set color for unfilled zone.
```
Comment (iRoachie) [inline docs/progressbar.md]:
```suggestion
Color of the unfilled area.
```

File: docs/progressbar.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+|      Type      | Default |
+| :------------: | :-----: |
+| Color (string) |  none   |
```
Comment (iRoachie) [inline docs/progressbar.md]:
```suggestion
| string |  none   |
```

File: src/progressbar/ProgressBar.js [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+  static defaultProps = {
+    containerStyle: {},
+    indeterminate: false,
+    progress: 0,
+    theme: {
+      colors: {
+        primary: 'blue',
+      },
+    },
```
Comment (iRoachie) [inline src/progressbar/ProgressBar.js]: Remove this. The theme will pass in the colors

File: src/progressbar/ProgressBar.js [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+
+class ProgressBarElement extends Component {
+  static propTypes = {
+    color: PropTypes.string,
+    indeterminate: PropTypes.bool,
+    progress: PropTypes.number,
+    style: PropTypes.any,
+    containerStyle: PropTypes.any,
+    unfilledColor: PropTypes.string,
+  };
```
Comment (iRoachie) [inline src/progressbar/ProgressBar.js]: Move this outside of the component at the bottom. That's the convention here

File: src/progressbar/ProgressBar.js (hunk 10 above)
Comment (iRoachie) [reply src/progressbar/ProgressBar.js]: Also move defaultProps out of the body

File: src/progressbar/ProgressBar.js [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+    };
+
+    return (
+      <View
+        style={[containerStyle, style]}
+        onLayout={this.handleLayout}
+        {...restProps}
+      >
+        <Animated.View style={progressStyle} />
+        {children}
```
Comment (iRoachie) [inline src/progressbar/ProgressBar.js]: Why is children here? Why should this accept children?

File: src/progressbar/ProgressBar.js [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+            inputRange: [0, 1],
+            outputRange: [0.0001, 1],
+          }),
+        },
+      ],
+    };
+
+    return (
+      <View
+        style={[containerStyle, style]}
```
Comment (iRoachie) [inline src/progressbar/ProgressBar.js]: If `containerStyle` and `style` are being applied on the same element then we don't need both.

In my thinking, we can have `containerStyle` and `progressStyle`

**Comment by sergeymorkovkin** [comment]
Is there any chance to get this done shortly?

**Comment by flyingcircle** [comment]
If someone can clean this up, we can get this into v3 soon.  I simplified the example app today. So please add a PR there with an example page showing off this new feature! Then we can make sure we tie up all the loose ends on the code and documentation.

**Comment by mateosilguero** [comment]
i can take it. will commit requested changes soon 👍

**Comment by flyingcircle** [comment]
@mateosilguero have you been able to make any progress on this? If you can also open a PR in the `react-native-elements-app` repo to demonstrate an example or two would be good too. Thanks.

**Comment by mateosilguero** [comment]
Good ! im making an example on that repo

**Pull request closed without merging**